zebra-state/src/service.rs

```diff
@@ -803,6 +803,91 @@ impl ReadStateService {
     fn latest_non_finalized_state(&self) -> NonFinalizedState {
         self.non_finalized_state_receiver.cloned_watch_data()
     }
+
+    /// Check the contextual validity of a block in the best chain
+    #[cfg(feature = "getblocktemplate-rpcs")]
+    #[allow(clippy::unwrap_in_result)]
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn check_best_chain_contextual_validity(
+        &self,
+        prepared: &PreparedBlock,
+    ) -> Result<(), crate::ValidateContextError> {
+        let latest_non_finalized_state = self.latest_non_finalized_state();
+
+        let Some(best_chain) = latest_non_finalized_state.best_chain() else {
+            Err(crate::ValidateContextError::NotReadyToBeValidated)?
+        };
+
+        if best_chain.non_finalized_tip_hash() != prepared.block.header.previous_block_hash {
+            Err(crate::ValidateContextError::NotReadyToBeValidated)?
+        }
+
+        check::initial_contextual_validity(
+            self.network,
+            &self.db,
+            &latest_non_finalized_state,
+            prepared,
+        )?;
```

Comment:
Does this check the best chain?

Reply:
Not anymore since it uses the regular `write::validate_and_commit_non_finalized` fn in [`8420b75`](https://github.com/ZcashFoundation/zebra/pull/5870/commits/8420b75013c1a05ea11a9799d1f87bd338c3c27c), but I think it was a mistake to do that check in the first place since it's not part of the server's usual acceptance rules.